src/main/java/sirius/kernel/xml/Outcall.java

```diff
@@ -626,4 +639,112 @@ public HttpRequest getRequest() {
     public static Average getTimeToFirstByte() {
         return timeToFirstByte;
     }
+
+    /**
+     * Checks to see if a new request is needed and builds it into the active {@link #request}.
+     */
+    private boolean checkAndBuildRedirectRequest(HttpResponse<InputStream> response) throws IOException {
+        if (redirectPolicy == HttpClient.Redirect.NEVER) {
+            // redirect disabled, no further checks needed
+            return false;
+        }
+        int responseCode = response.statusCode();
+        if (isRedirecting(responseCode)) {
+            URI redirectedURI = makeRedirectedURI(response.headers());
+            if (canRedirect(redirectedURI)) {
+                // copy and modify the base request builder
+                HttpRequest.Builder redirectBuilder = requestBuilder.copy();
+                redirectBuilder.uri(redirectedURI);
+                if (shouldSwitchToGet(responseCode, request.method())) {
+                    redirectBuilder.GET();
+                }
+                request = redirectBuilder.build();
+                return true;
+            } else {
+                return false;
+            }
+        }
+        return false;
+    }
+
+    private boolean isRedirecting(int statusCode) {
+        if (statusCode < 300) {
+            // < 300: not a redirect code
+            return false;
+        }
+        if (statusCode > 308) {
+            // 309-399 Unassigned => don't follow
+            // > 399: not a redirect code
+            return false;
+        }
+
+        return switch (statusCode) {
+            // 300: MultipleChoice => don't follow
+            // 304: Not Modified => don't follow
+            // 305: Proxy Redirect => don't follow.
+            // 306: Unused => don't follow
+            case 300, 304, 305, 306 -> false;
+            // 301: Moved permanently => follow
+            // 302: Found => follow
+            // 303: See other => follow
+            // 307: Temp Redirect => follow
+            // 308: Permanent Redirect => follow
+            default -> true;
+        };
+    }
```

Comment:
Couldn't this be way simpler with an if code == 301 || code == 302 || code == 303 || code == 307 || code == 308

Also URLConnection provides constants for all codes..... (Maybe the HttpClient does as well....)

Reply:
or put these codes in a list and check with contains?